neurostack/data_streams/data_stream.py

```diff
@@ -1,38 +1,105 @@
+"""
+LSL-like object to stream data to multiple channels. Some code and helper
+methods taken from https://github.com/kaczmarj/rteeg
+"""
+import numpy as np
+import pylsl
+
+
+def look_for_eeg_stream():
+    """returns an inlet of the first eeg stream outlet found."""
+    print("looking for an EEG stream...")
+    streams = pylsl.resolve_byprop('type', 'EEG', timeout=30)
+    if len(streams) == 0:
+        raise (RuntimeError, "Can't find EEG stream")
+    print("Start acquiring data")
+    eeg_inlet = pylsl.StreamInlet(streams[0], max_chunklen=1)
+
+    return eeg_inlet
+
+
 class DataStream:
 
     def __init__(self):
         """Initializes data stream"""
         self.channels = {}
+        self.thread = None
+        self.eeg_channel_names = None
 
     #
     # Connection methods
     #
 
     def lsl_connect(self):
         """Connects to LSL stream"""
-        pass
+        # get stream
+        eeg_stream = look_for_eeg_stream()
+        info = eeg_stream.info()
+
+        # get channel names
+        ch_names = []
+        this_child = info.desc().child('channels').child('channel')
+        for _ in range(info.channel_count()):
+            ch_names.append(this_child.child_value('label'))
+            this_child = this_child.next_sibling('channel')
+        self.eeg_channel_names = ch_names
+
+        # record data to channels
+        self._thread = threading.Thread(target=self._record_lsl_data_indefinitely,
```

Comment:
Need to initialize threading before using

Reply:
Fixed